src/Cli/ConfigGenerator.cs

```diff
@@ -127,7 +137,10 @@ public static bool TryCreateRuntimeConfig(InitOptions options, out string runtim
                 return false;
             }
 
-            if (!IsApiPathValid(restPath, ApiType.REST) || !IsApiPathValid(options.GraphQLPath, ApiType.GraphQL))
+            if (!IsURIComponentValid(restPath, ApiType.REST, ApiSettings.PROPERTY_NAME_PATH) ||
+                !IsURIComponentValid(options.GraphQLPath, ApiType.GraphQL, ApiSettings.PROPERTY_NAME_PATH) ||
+                !IsURIComponentValid(restBaseRoute, ApiType.REST, ApiSettings.PROPERTY_NAME_BASE_ROUTE))
+
```

Comment:
I wonder if we'd be better splitting this into three separate `if` blocks so that we can provide better diagnostic information as to why something failed to pass.

Reply:
This will already be logged here: https://github.com/Azure/data-api-builder/blob/b818d6fa239ebb80ed71b9bfa335924c4da8d819/src/Cli/Utils.cs#L213.